### PR TITLE
aspects: improve errors caused by unexpected types

### PR DESCRIPTION
**Pull request by MiguelPires**
Improve errors caused by unexpected types when validating against the schema.

### CODE REVIEW

File: aspects/schema.go [hunk 1]
```diff
@@ -556,6 +564,10 @@ func (v *intSchema) Validate(raw []byte) (err error) {
 
 	var num *int64
 	if err := json.Unmarshal(raw, &num); err != nil {
+		typeErr := &json.UnmarshalTypeError{}
+		if errors.As(err, &typeErr) {
+			return fmt.Errorf("expected int type but got %s", typeErr.Value)
```
Comment (andrewphelpsj) [inline aspects/schema.go:569]: Why `validationErrorf` in some places, but `fmt.Errorf` in other places?

File: aspects/schema_test.go [hunk 2]
```diff
@@ -1892,3 +1829,60 @@ func (*schemaSuite) TestValidationError(c *C) {
...
+		schemaStr := []byte(fmt.Sprintf(`{
+	"schema": {
+		"foo": %s
+	}
+}`, tc.schemaType))
+		schema, err := aspects.ParseSchema(schemaStr)
+		c.Assert(err, IsNil)
+
+		var testValue, testType string
+		if tc.expectedType != "bool" {
```
Comment (andrewphelpsj) [inline aspects/schema_test.go:1876]: nitpick: I think the test would be a little easier to read at the cost of some duplication if this `if` were removed and `testValue` and `testType` were put in each `testcase`

**Review by andrewphelpsj — APPROVED** [review]
Thanks! Just some tiny things.

File: aspects/schema_test.go (hunk 2 above)
Comment (MiguelPires) [reply aspects/schema_test.go]: I removed the if/else and added an `interface{}` field so we can get both the value and type from it. Thanks

**Review by pedronis — APPROVED** [review]
thank you

**Pull request merged**